compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -12438,7 +12438,9 @@ private boolean isEndOfGroupByListOrListElement(SyntaxKind tokenKind) {
             case EOF_TOKEN:
                 return true;
             default:
-                return isQueryClauseStartToken(tokenKind);
+                // First case is a special case to improve group by key list end parsing.
+                return peek(2).kind == SyntaxKind.IDENTIFIER_TOKEN && peek(3).kind != SyntaxKind.EQUAL_TOKEN ||
```

Comment:
is checking `peek(3).kind != SyntaxKind.EQUAL_TOKEN` correct? We have a capture or wildcard binding pattern after the identifier?

Reply:
Yes we are dealing with the `T variableName = expression` case here, where `T` is the identifier on `peek(1)`

Reply:
My bad. the `tokenKind` argument passing into the `isEndOfGroupByListOrListElement()` is `getNextNextToken().kind`. We can use it directly instead of `peek(2).kind`.

Reply:
But we need `peek(3)` here right?

Reply:
Updated in [47ecc38](https://github.com/ballerina-platform/ballerina-lang/pull/40297/commits/47ecc38878763b9b7a35eb74eff7d9843ba81fcc)